### PR TITLE
deprecate `odbcConnectionColumns()`

### LINKED ISSUE #699: `odbcConnectionTableTypes()` is not exported
`odbcConnectionTableTypes()` has a man-page:

https://github.com/r-dbi/odbc/blob/95e8b69833fabcbc78888a917e83cb0e9af78ddb/R/odbc-connection.R#L427-L429

and some exported functions refer to it in theirs:

https://github.com/r-dbi/odbc/blob/95e8b69833fabcbc78888a917e83cb0e9af78ddb/R/odbc-connection.R#L297

but the function isn't exported.

Is this on purpose?

### PR DESCRIPTION
**Pull request by simonpcouch**
Closes #699.

Note that this PR doesn't touch the `odbcConnectionActions` and `odbcConnectionIcon` functions, which are also exported.

### CODE REVIEW

File: R/odbc-connection.R [hunk 1]
```diff
@@ -140,11 +140,15 @@ NULL
 
 #' odbcConnectionColumns
 #'
+#' @description
+#' `r lifecycle::badge("deprecated")`
```
Comment (hadley) [inline R/odbc-connection.R:144]: I'd suggest you recommend `dbListFields()` here too.

File: R/odbc-connection.R [hunk 2]
```diff
@@ -176,21 +180,33 @@ NULL
 #'
 #' @rdname odbcConnectionColumns
```
Comment (hadley) [inline R/odbc-connection.R:181]: I think you could `@keywords internal` and make sure it's not include in the pkgdown ref too.

**Review by hadley — APPROVED** [review]
This approach looks good to me. I think you could also do `odbcConnectionActions()` and `odbcConnectionIcon()` in this PR too if you wanted.

File: R/odbc-connection.R (hunk 2 above)
Comment (simonpcouch) [reply R/odbc-connection.R:192]: Sounds good! 5096a4b.

File: R/odbc-connection.R (hunk 1 above)
Comment (simonpcouch) [reply R/odbc-connection.R:144]: You got it! 9abd87a.

**Comment by simonpcouch** [comment]
In deprecating `odbcConnectionActions()` in 127387a I didn't use the temporary underscore pattern as we make no "internal" usage of the function. It doesn't look me to like this is called anywhere in the IDE, so it seems fine to deprecate as usual.

**Comment by simonpcouch** [comment]
Same goes for [odbcConnectionIcon()](https://github.com/r-dbi/odbc/pull/750/commits/73aa7e252913a7e9784dd18f6b83e8d6f301ea44).

**Comment by hadley** [comment]
Aren't they both called in `on_connection_opened()`?

**Comment by simonpcouch** [comment]
Ah, I missed that somehow. Back to it.🐢

**Comment by simonpcouch** [comment]
Transitioned `odbcConnectionActions()` and `odbcConnectionIcon()` from S3 methods, as they only have a default method and our change to the underscore naming would break dispatch to any methods supplied by other packages anyway.

**Review by hadley — APPROVED** [review]

**Pull request merged**